include/lsst/jointcal/AstrometryFit.h

```diff
@@ -52,13 +52,10 @@ class Associations;
  * CcdImage's. One does not need reference object and would then naturally not
  * have any Kind 2 terms.
  */
-class AstrometryFit {
+class AstrometryFit : public FitterBase {
 private:
-    Associations &_assoc;
-    std::string _whatToFit;
     bool _fittingDistortions, _fittingPos, _fittingRefrac, _fittingPM;
-    AstrometryModel *_astrometryModel;
-    int _LastNTrip;                   // last triplet count, used to speed up allocation
+    AstrometryModel &_astrometryModel;
```

Comment:
After some offline discussion, we decided that shared_ptr was appropriate for this case: the `Associations` instance outlives any given fitter, and the `Model`s must live at least as long as the fitter that works with them.